sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -2820,6 +2821,23 @@ object SQLConf {
       .booleanConf
       .createWithDefault(true)
 
+  object TimestampTypes extends Enumeration {
+    val TIMESTAMP_NTZ, TIMESTAMP_LTZ = Value
+  }
+
+  val TIMESTAMP_TYPE =
+    buildConf("spark.sql.timestampType")
+      .doc("Configures the default timestamp type of Spark SQL, including SQL DDL and Cast " +
```

Comment:
I wonder about types literals, for instance `timestamp'2021-07-01 01:02:03'`. Should the config influence on it too?

Reply:
Yes, I mentioned in https://github.com/apache/spark/pull/33176#discussion_r662426271. The type literal logic with TIMESTAMP_NTZ:
1. if there is no time zone part, return timestamp without time zone literal
2. otherwise, return timestamp with local time zone.